addons/mail/static/src/models/emoji_category_view.js

```diff
@@ -1,21 +1,61 @@
 /** @odoo-module **/
 
 import { registerModel } from '@mail/model/model_core';
-import { one } from '@mail/model/model_field';
+import { attr, one } from '@mail/model/model_field';
+import { replace } from '@mail/model/model_field_command';
 
 registerModel({
     name: 'EmojiCategoryView',
-    identifyingFields: ['emojiCategory', 'emojiPickerView'],
+    identifyingFields: ['emojiCategory', 'emojiCategoryBarViewOwner'],
+    recordMethods: {
+        /** 
+         * @param {MouseEvent} ev
+         */
+        onClick() {
+            this.update({ emojiCategoryBarViewOwnerAsActive: replace(this.emojiCategoryBarViewOwner) });
+        },
+        /**
+         * @param {MouseEvent} ev
+         */
+        onMouseenter(ev) {
+            this.update({ isHover: true });
+        },
+        /**
+         * @param {MouseEvent} ev
+         */
+        onMouseleave(ev) {
+            this.update({ isHover: false });
+        },
+        /**
+         * @private
+         * @returns {boolean}
+         */
+        _computeIsActive() {
+            if (this.emojiCategoryBarViewOwnerAsActive) {
+                return true;
+            }
+            return false;
+        },
+    },
     fields: {
         emojiCategory: one('EmojiCategory', {
             inverse: 'emojiCategoryViews',
             readonly: true,
             required: true,
         }),
-        emojiPickerView: one('EmojiPickerView', {
+        emojiCategoryBarViewOwner: one('EmojiCategoryBarView', {
             inverse: 'emojiCategoryViews',
             readonly: true,
             required: true,
         }),
+        emojiCategoryBarViewOwnerAsActive: one('EmojiCategoryBarView', {
```

Comment:
It's not in identifying field, but it's very important that the one being related to is exactly the same as owner, so that it's semantically correct. That's why the "owner" is very important in the name.

Reply:
OK for the owner part (even though we have nothing to enforce what you said), but the part after "as" should ideally be the full name of the inverse field `activeCategoryView` in this case. With just "AsActive" we don't know what is actually supposed to be active, is it the owner? the current record?

Reply:
In the pattern with `X as Y`, `X` always refer to the other, while the `Y` always refer to current. So here it's current being active.